pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -170,19 +171,42 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			}
 		}
 		// fire instance handles for workload
-		ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(ip, 0, "", model.AlwaysDiscoverable)
-		workloadInstance := &model.WorkloadInstance{
-			Name:      pod.Name,
-			Namespace: pod.Namespace,
-			Kind:      model.PodKind,
-			Endpoint:  ep,
-			PortMap:   getPortMap(pod),
+		if len(pc.c.handlers.GetWorkloadHandlers()) > 0 {
```

Comment:
it looks redundant, isn't this  always > 0

Reply:
I'm not sure, shouldn't it be 0 if `ServiceEntrySelectPod` is disabled?

Reply:
it is still >0, just the handler does nothing

Reply:
https://github.com/istio/istio/blob/1202dc2a681b4b6a9e98f5414249f346129a2188/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L177-L180

Why? Is there any other place where appending workload handlers?

Reply:
right